### DISCUSSION TITLE
Always set react.development=true when in jest mode ?

### DISCUSSION THREAD

**Discussion by stefee** [swc-project/swc#2239] [Q&A] [answered]
Is there ever a reason not to set react.development to true when in jest mode? It seems like you would always want this mode enabled for diagnostics purposes when a test failure occurs. Curious to know what the trade-offs are exactly.

CC @Brooooooklyn as this relates to `@swc-node/jest` as well as `@swc/jest`

**Comment by kdy1** [comment, marked as answer]
I think this is a good change, but I'm not sure if it should be handled from rust code.

**Reply by stefee** [reply]
My sense is this should stay as a configuration the same as now but either swc/jest and/or swc-node/jest should default it.

I’m happy to open a PR in both repos and we can evaluate impact from there.

**Reply by kdy1** [reply]
Yeah, I think you are right.
Can you send a PR?

**Reply by stefee** [reply]
Perhaps we should confirm whether or not jest tooling makes use of the `__self` and `__source` props, or if it only used for React Developer Tools browser extension.

(Either way there is no harm in enabling it, but minor additional code bloat)
